scripts/load-test/collect-pod-logs/collect-openshift-pod-logs.sh

```diff
@@ -0,0 +1,168 @@
+      if [ "$event" == "ADDED" ]; then
+        echo "New pod added: $pod_name in namespace $namespace"
+        pod_dir="$namespace_dir/$pod_name"
+        mkdir -p "$pod_dir"
+        # Watch for added or deleted containers in the new pod
+        kubectl get pods "$pod_name" --watch --namespace "$namespace" --output-watch-events --output jsonpath='{.type} {.object.spec.containers[*].name}{"\n"}' | while read -r event container_name; do
+          if [ "$event" == "ADDED" ]; then
+            echo "New container added: $container_name in pod $pod_name in tenant namespace $namespace"
+            container_dir="$pod_dir"
+            collect_logs "${namespace}" "${pod_name}" "${container_name}" "${container_dir}"
+          elif [ "$event" == "DELETED" ]; then
+            echo "Container deleted: $container_name in pod $pod_name in tenant namespace $namespace"
+            list_log_file "${namespace}" "${pod_name}" "${container_name}"
+          fi
+        done &
+      elif [ "$event" == "DELETED" ]; then
+        echo "Pod deleted: $pod_name in tenant namespace $namespace"
+      fi
+    done &
+  done
+}
+
+
+collect_logs_from_new_namespaces() {
+  while true; do
+    kubectl get namespaces --watch --output-watch-events | while read -r event; do
+      event_type=$(echo "$event" | awk '{print $1}')
+      namespace=$(echo "$event" | awk '{print $2}')
+
+      if [ "$event_type" == "ADDED" ] && echo "$namespace" | grep -q ".*-tenant$"; then
+      # Searching for tenant named namespaces
+          echo "New tenant namespace added: $namespace"
+          namespace_dir="$log_dir/$namespace"
+          mkdir -p "$namespace_dir"
+
+          # Watch for added or deleted pods in the new tenant namespace
+          kubectl get pods --watch --namespace "$namespace" --output-watch-events --output jsonpath='{.type} {.object.metadata.name}{"\n"}' | while read -r event pod_name; do
+            if [ "$event" == "ADDED" ]; then
+              echo "New pod added: $pod_name in tenant namespace $namespace"
+              pod_dir="$namespace_dir/$pod_name"
+              mkdir -p "$pod_dir"
+
+              # Watch for added or deleted containers in the new pod
+              kubectl get pods "$pod_name" --watch --namespace "$namespace" --output-watch-events --output jsonpath='{.type} {.object.spec.containers[*].name}{"\n"}' | while read -r event container_name; do
+                if [ "$event" == "ADDED" ]; then
+                  echo "New container added: $container_name in pod $pod_name in tenant namespace $namespace"
+                  container_dir="$pod_dir"
+                  collect_logs "${namespace}" "${pod_name}" "${container_name}" "${container_dir}"
+                elif [ "$event" == "DELETED" ]; then
+                  echo "Container deleted: $container_name in pod $pod_name in tenant namespace $namespace"
+                  list_log_file "${namespace}" "${pod_name}" "${container_name}"
+                fi
+              done &
```

Comment:
@naftalysh If a pod has more than one container, then the watch command produces output like:
```
ADDED step-build step-sbom-get step-analyse-dependencies-java-sbom step-merge-sboms step-inject-sbom-and-push step-upload-sbom
```
with the list of all containers considered as a name of the container which is incorrect.

In such case the captured log file is like https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_release/37577/rehearse-37577-periodic-ci-redhat-appstudio-e2e-tests-main-load-test/1640992270981271552/artifacts/load-test/redhat-appstudio-load-test/artifacts/testuser-0001-tenant/testuser-0001-component-8dc9d-build-container-pod/step-build%20step-sbom-get%20step-analyse-dependencies-java-sbom%20step-merge-sboms%20step-inject-sbom-and-push%20step-upload-sbom.log with the content of

```
error: container step-build step-sbom-get step-analyse-dependencies-java-sbom step-merge-sboms step-inject-sbom-and-push step-upload-sbom is not valid for pod testuser-0001-component-8dc9d-build-container-pod
```

You need to further split the "container_name" into individual containers - something like this:
```suggestion
              kubectl get pods "$pod_name" --watch --namespace "$namespace" --output-watch-events --output jsonpath='{.type} {.object.spec.containers[*].name}{"\n"}' | while read -r event container_names; do
                while read -d " " -r container_name; do
                  if [ "$event" == "ADDED" ]; then
                    echo "New container added: $container_name in pod $pod_name in tenant namespace $namespace"
                    container_dir="$pod_dir"
                    collect_logs "${namespace}" "${pod_name}" "${container_name}" "${container_dir}"
                  elif [ "$event" == "DELETED" ]; then
                    echo "Container deleted: $container_name in pod $pod_name in tenant namespace $namespace"
                    list_log_file "${namespace}" "${pod_name}" "${container_name}"
                  fi
                done <<< $container_names
              done &
```

Please apply this change in the other part of the script, as well. Thanks!

Reply:
Hi @pmacik 
Thanks for the suggestion, I used something similar
Now it iterates correctly the containers